src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,169 @@
+            add(sample);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index " + index + " out of bound ");
+        }
+        Node<T> currentNode = first;
+        for (int i = 0; i < index; i++) {
+            currentNode = currentNode.next;
+        }
+        return currentNode.element;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index " + index + " out of bound ");
+        }
+        Node<T> currentNode = first;
+        for (int i = 0; i < index; i++) {
+            currentNode = currentNode.next;
+        }
+        T oldValue = currentNode.element;
+        currentNode.element = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index " + index + " out of bound ");
+        }
+        Node<T> nodeToRemove;
+        if (index == 0) {
+            nodeToRemove = first;
+            first = first.next;
+            if (first != null) {
+                first.previous = null;
+            }
+            if (first == null) {
+                last = null;
+            }
+        } else if (index == size - 1) {
+            nodeToRemove = last;
+            last = last.previous;
+            if (last != null) {
+                last.next = null;
+            }
+            if (last == null) {
+                first = null;
+            }
+        } else {
+            nodeToRemove = first;
+            for (int i = 0; i < index; i++) {
+                nodeToRemove = nodeToRemove.next;
+            }
+            nodeToRemove.previous.next = nodeToRemove.next;
+            nodeToRemove.next.previous = nodeToRemove.previous;
+        }
+        size--;
+        return nodeToRemove.element;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> currentNode = first;
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.element, object)) {
+                if (currentNode == first) {
+                    first = first.next;
+                    if (first != null) {
+                        first.previous = null;
+                    }
+                } else if (currentNode == last) {
+                    last = last.previous;
+                    if (last != null) {
+                        last.next = null;
+                    }
+                } else {
+                    currentNode.previous.next = currentNode.next;
+                    currentNode.next.previous = currentNode.previous;
+                }
+                size--;
+                return true;
+            }
+            currentNode = currentNode.next;
+        }
         return false;
     }
```

Comment:
Create a separate method `unlink(Node node)` for removing a node from the list to avoid duplicating the unlinking logic. Also, ensure that you consider `null` as a method parameter in the `remove(T object)` method.